source/playbooks/CIS/lambda/cis120.py

```diff
@@ -0,0 +1,206 @@
+        """
+        Send Failed status message
+        """
+        message['State'] = 'FAILED'
+        message['Note'] = ''
+        notify(finding, message, LOGGER, cwlogs=APPLOGGER)
+
+    # Make sure it matches - custom action can be initiated for any finding.
+    # Ignore if the finding selected and the playbook do not match
+    cis_data = finding.is_cis_ruleset()
+    if not cis_data:
+        # Not an applicable finding - does not match ruleset
+        # send an error and exit
+        LOGGER.debug('CIS 1.20: incorrect custom action selection')
+        APPLOGGER.add_message('CIS 1.20: incorrect custom action selection')
+        return
+
+    if (cis_data['ruleid'] not in ['1.20']):
+        # Not an applicable finding - does not match rule
+        # send an error and exit
+        LOGGER.debug('CIS 1.20: incorrect custom action selection')
+        APPLOGGER.add_message('CIS 1.20: incorrect custom action selection')
+        return
+
+    #==========================================================================
+    message['AffectedObject'] = AFFECTED_OBJECT
+    try:
+        sess = BotoSession(finding.account_id, LAMBDA_ROLE)
+        iam = sess.client('iam')
+    except Exception as e:
+        LOGGER.error(e)
+        failed()
+        return
+
+    # Mark the finding NOTIFIED while we remediate
+    message['State'] = 'INITIAL'
+    notify(finding, message, LOGGER, cwlogs=APPLOGGER)
+
+    try:
+        resource = finding.details['Resources'][0]
+        id_parts = resource['Id'].split(':')
+        account = id_parts.pop()
+
+        aws_support_policy = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Action": "sts:AssumeRole",
+                    "Principal": {
+                        "AWS": f"arn:aws:iam::{account}:root"
+                    }
+                }
+            ]
+        }
+
+        role_name = 'aws_incident_support_role'
+
+        if does_role_exist(iam, role_name):
+            lowercase_str = uuid.uuid4().hex
+            role_name = f'{role_name}_{lowercase_str[0:8]}'
+
+        iam.create_role(
+            RoleName=role_name,
+            AssumeRolePolicyDocument=json.dumps(aws_support_policy),
+            Description='Created by SHARR security hub remediation 1.20 rule',
+            Tags=[
+                {
+                    'Key': 'Name',
+                    'Value': 'CIS 1.20 aws support access role'
+                },
+            ]
+        )
+
+        iam.attach_role_policy(
+            RoleName=role_name,
+            PolicyArn='arn:aws:iam::aws:policy/AWSSupportAccess',
+        )
+
+        message['State'] = 'RESOLVED'
+        message['Note'] = '' # use default
+        notify(finding, message, LOGGER, cwlogs=APPLOGGER, sns=AWS)
+        return
+
+    except Exception as e:
+        LOGGER.error(e)
+        failed()
+        return
+
+
+def does_role_exist(iam, role_name):
```

Comment:
This function needs to go above where it is called or you will get a `NameError` not found.

Reply:
I doubt if this is the case, as in Python you can define in any order. did you confuse with JS? I tested both locally and playbook actions.

Reply:
I swear it's python, but no probs. (maybe it used to be the case...)